pkg/apis/elasticsearch/v1alpha1/types.go

```diff
@@ -109,4 +109,15 @@ type ElasticsearchStatus struct {
 	// Fill me
 	Nodes         []ElasticsearchNodeStatus `json:"nodes"`
```

Comment:
Do we think we can do away with this with the inclusion of `Pods`?
With this PR this is what status will look like currently:
```
status:
  clusterHealth: cluster health unknown
  nodes:
  - deploymentName: elasticsearch-clientdatamaster-0-1
    podName: elasticsearch-clientdatamaster-0-1-79d5c457db-gzz49
    replicaSetName: elasticsearch-clientdatamaster-0-1-79d5c457db
    roles:
    - client
    - data
    - master
    status: Running
  pods:
    client:
      failed: []
      notReady:
      - elasticsearch-clientdatamaster-0-1-79d5c457db-gzz49
      ready: []
    data:
      failed: []
      notReady:
      - elasticsearch-clientdatamaster-0-1-79d5c457db-gzz49
      ready: []
    master:
      failed: []
      notReady:
      - elasticsearch-clientdatamaster-0-1-79d5c457db-gzz49
      ready: []
```

Reply:
Yes - looks like `pods` isn't necessary - would much rather see the information in the format as displayed under `nodes` - but the `nodes[0].status` should be `notReady`?

Reply:
@richm the purpose of this PR is to introduce `pods` as that provides information that can be consumed and displayed on the console as part of the OLM work.

To me it makes more sense to see it displayed like that rather than have `podName` under `nodes` and display the `roles` there too

Reply:
the `nodes` output is more human readable than the `pods` output, but if the intention is that the `pods` output is not for humans, then ok

Reply:
@ewolinetz what format doe the OLM expect?

To explain a bit current format of `.status`:
We now deploy 1 Pod per 1 Deployment and potentially 1 PV, which is why 1 ES node = 1 Deployment = 1 Pod.
That's why `.status.nodes.*.status` is `pod.status.phase`

Reply:
@josefkarasek we can specify a path for OLM, however in order to use 
```
x-descriptors:
        - 'urn:alm:descriptor:com.tectonic.ui:podStatuses'
```
We need to be in the format of `map[string][]string` for our pods so that it can be displayed on the console similar to this [1] if you scroll down to the UI example.

[1] https://github.com/openshift/console/tree/master/frontend/public/components/operator-lifecycle-manager/descriptors#example

Reply:
my only concern is that there should not be `pods.client`, `pods.data` and `pods.master` - there should be `pods.failed`, `pods.ready` and `pods.notReady` - but I'm not sure where to indicate that a particular pod is a client, data, and/or master